test/pin.js

```diff
@@ -22,140 +22,139 @@ describe('pin', function () {
   }]
 
   let daemons = []
-  function spawnAndStart (type, repoPath = utils.tmpPath()) {
-    return new Promise((resolve, reject) => {
-      DaemonFactory.create({ type })
-        .spawn({
-          repoPath,
-          disposable: false
-        }, (err, daemon) => {
-          if (err) return reject(err)
-          daemons.push(daemon)
-
-          if (daemon.initialized) {
-            // repo already exists, no need to init
-            daemon.start(err => err ? reject(err) : resolve(daemon))
-          } else {
-            daemon.init((err, initRes) => {
-              if (err) return reject(err)
-              daemon.start(err => err ? reject(err) : resolve(daemon))
-            })
-          }
-        })
-    })
+  async function spawnAndStart (type, repoPath = utils.tmpPath()) {
+    const daemonOptions = {
+      repoPath,
+      disposable: false
+    }
+
+    const daemon = await (type === 'go' ? spawnInitAndStartGoDaemon(daemonOptions) : spawnInitAndStartJsDaemon(daemonOptions))
+    daemons.push(daemon)
+
+    if (daemon.initialized) {
```

Comment:
If we do `spawnInitAndStartGoDaemon` / `spawnInitAndStartJsDaemon`, this logic for init/start should just go away?

Reply:
I'll rename these functions' names to `spawnGoDaemon` and `spawnJsDaemon` and this way everything is more intuitive.